### PR TITLE
implementation of lookup for playlists in library, and track caching

### PR DESCRIPTION
**Pull request by MrSurly**
This is what I hope is a beginning of a fix for #15.  In any case, I talked to the mopidy devs about the issue, and implementing this is probably desirable.

I think a fix for #15 will require changes to Iris.

### CODE REVIEW

**Comment by MrSurly** [comment]
@tehkillerbee This works for me -- the Mopidy devs recommended adding the track caching from the `lookup` calls.

**Comment by tehkillerbee** [comment]
@MrSurly Great to see you have had progress! Looking forward to test it during the next couple of days.

File: mopidy_tidal/library.py [hunk 1]
```diff
@@ -18,6 +18,24 @@
 
 logger = logging.getLogger(__name__)
 
+class Cache(dict): 
+    '''cache class, requires python3 style (ordered) dicts'''
```
Comment (kingosticks) [inline mopidy_tidal/library.py:22]: If this was a an actual `OrderedDict` you could use [`popitem(last=False)`](https://docs.python.org/3.7/library/collections.html#collections.OrderedDict.popitem) to simplify your `__prune()`.

Comment (MrSurly) [reply mopidy_tidal/library.py:22]: I'm relying on Python 3 dicts (which are essentially `OrderedDict`).  For some reason, I thought Mopidy was Python3 only?

I'll get this changed in any case.

Comment (MrSurly) [reply mopidy_tidal/library.py:25]: Updated.

Comment (kingosticks) [reply mopidy_tidal/library.py:25]: You are totally correct that Mopidy requires Python 3.7+ and in Python 3.7, dict became ordered. But dict doesn't have all the same functionality as `OrderedDict`. Popping from the front with `last=False` is only possible with  `OrderedDict`.

Comment (MrSurly) [reply mopidy_tidal/library.py:25]: I saw that, which is why I changed it.

**Comment by tehkillerbee** [comment]
@MrSurly @kingosticks Nice work, it looks like the caching works as expected - playback starts quite fast now. I still think it would be nice to get some sort of image caching added too (similar to the suggested cache changes in #27), as I believe it would help speeding up the browsing experience too. I will merge these changes in now.

**Pull request merged**